File: sdk-node/examples/events/error-handling.ts
/**
 * Demonstrate the SDK's typed error hierarchy. Every error thrown by the SDK
 * is a subclass of `ThreeCommonError`.
 *
 * Run:
 *   npx tsx examples/events/error-handling.ts
 */

import {
  ThreeCommon,
  ThreeCommonAuthError,
  ThreeCommonNotFoundError,
  ThreeCommonRateLimitError,
} from '@3-common/sdk'

const client = new ThreeCommon({
  apiKey: '3co_your_api_key_here',
})

try {
  const event = await client.events.retrieve('evt_definitely_does_not_exist_12345')
  console.log(event)
} catch (err) {
  if (err instanceof ThreeCommonNotFoundError) {
    console.log('Not found:', err.message, '(request_id:', err.requestId, ')')
  } else if (err instanceof ThreeCommonAuthError) {
    console.log('Bad or expired API key:', err.message)
  } else if (err instanceof ThreeCommonRateLimitError) {
    console.log('Rate limited; retry after', err.retryAfterSeconds, 'seconds')
  } else {
    throw err
  }
}

File: sdk-python/examples/events/error_handling.py
"""Run with: python examples/events/error_handling.py"""

from __future__ import annotations

from threecommon import (
    AuthError,
    ConnectionError,
    NotFoundError,
    RateLimitError,
    ThreeCommon,
)


def main() -> None:
    with ThreeCommon(api_key="3co_your_api_key_here") as client:
        try:
            client.events.retrieve("000000000000000000000000")
        except NotFoundError as e:
            print(f"event not found — request_id={e.request_id}")
        except AuthError as e:
            print(f"auth failed: check your API key — code={e.code}")
        except RateLimitError as e:
            wait = e.retry_after_seconds or 30
            print(f"rate limited; waiting {wait}s before retry")
        except ConnectionError as e:
            print(f"network error: {e.__cause__}")


if __name__ == "__main__":
    main()
